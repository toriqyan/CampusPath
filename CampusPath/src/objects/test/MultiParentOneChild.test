# Tests that a graph can be created, three node and
# four edges from different parents to the same child can be added, 
# and one node and four edges can be printed out.

# Create a graph
CreateGraph TheLordOfTheRing

# Add four nodes
AddNode TheLordOfTheRing Frodo
AddNode TheLordOfTheRing Gandalf
AddNode TheLordOfTheRing Samwise

# Add four edges
AddEdge TheLordOfTheRing Gandalf Frodo TheHobbit
AddEdge TheLordOfTheRing Samwise Frodo TheReturnOfTheKing
AddEdge TheLordOfTheRing Samwise Frodo TheTwoTowers
AddEdge TheLordOfTheRing Samwise Frodo TheFellowshipOfTheRing

# Print the nodes in the graph and the outgoing edges from Frodo, Gandalf and Samwise
ListNodes TheLordOfTheRing
ListChildren TheLordOfTheRing Frodo
ListChildren TheLordOfTheRing Gandalf
ListChildren TheLordOfTheRing Samwise