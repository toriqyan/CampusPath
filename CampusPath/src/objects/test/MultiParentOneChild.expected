# Tests that a graph can be created, three node and
# four edges from different parents to the same child can be added, 
# and one node and four edges can be printed out.

# Create a graph
created graph TheLordOfTheRing

# Add four nodes
added node Frodo to TheLordOfTheRing
added node Gandalf to TheLordOfTheRing
added node Samwise to TheLordOfTheRing

# Add four edges
added edge TheHobbit from Gandalf to Frodo in TheLordOfTheRing
added edge TheReturnOfTheKing from Samwise to Frodo in TheLordOfTheRing
added edge TheTwoTowers from Samwise to Frodo in TheLordOfTheRing
added edge TheFellowshipOfTheRing from Samwise to Frodo in TheLordOfTheRing

# Print the nodes in the graph and the outgoing edges from Frodo, Gandalf and Samwise
TheLordOfTheRing contains: Frodo Gandalf Samwise
the children of Frodo in TheLordOfTheRing are: 
the children of Gandalf in TheLordOfTheRing are: Frodo(TheHobbit)
the children of Samwise in TheLordOfTheRing are: Frodo(TheFellowshipOfTheRing) Frodo(TheReturnOfTheKing) Frodo(TheTwoTowers)